psh/rmdir/rmdir.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Phoenix-RTOS
+ *
+ * rmdir - remove empty directories
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Gerard Swiderski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "../psh.h"
+
+
+static void psh_rmdir_info(void)
+{
+	printf("remove empty directories");
+}
+
+
+static void psh_rmdir_usage(void)
+{
+	printf("Usage: rmdir [-p] DIRECTORY...\n");
+}
+
+
+static int removedir(char *name, int parents)
+{
+	char *end = name + strlen(name) - (size_t)1;
+	int suppress = 0;
+	int err = -1;
+
+	while (name[0] != '\0') {
+		while ((end >= name) && (*end == '/')) {
+			*(end--) = '\0';
+		}
+
+		if (name[0] != '\0') {
+			err = rmdir(name);
+			end = strrchr(name, '/');
+			if ((err != 0) || (end == NULL) || (parents == 0)) {
+				break;
+			}
+
+			suppress = 1;
+		}
+	}
+
+
+	return ((err != 0) && (suppress == 0));
+}
+
+
+static int psh_rmdir(int argc, char **argv)
+{
+	char *dirname;
+	int i, parent = 0, ret = EXIT_SUCCESS;
+
+	if ((argc > 1) && (argv[1][0] == '-') && (argv[1][1] == 'p')) {
```

Comment:
```suggestion
	if ((argc > 1) && (argv[1][0] == '-') && (argv[1][1] == 'p') && (argv[1][2] == '\0')) {
```
prevents, for example, such calling:
```
psh_rmdir -ptr dir
```

Reply:
Done.